lxd/archive/archive.go

```diff
@@ -0,0 +1,71 @@
+package archive
+
+import (
+	"archive/tar"
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/lxc/lxd/lxd/apparmor"
+	"github.com/lxc/lxd/lxd/state"
```

Comment:
I wondered why we needed the `state` package here, but it looks like its for the `apparmor` package.
Looking at the `apparmor` package, its dependency on `state` seems to be unnecessary, I did a quick scan through that package and from what I can see it really only depends on `state.OS` which is a `*sys.OS`. If you have time it would be great to put up a separate PR to remove the `state` dependency from `apparmor` and just have those functions accept a `*sys.OS` argument called `os`. Then in turn this PR can have its `state` dependency removed and just have a similar `os` argument. This way its clearer what these functions (and the apparmor ones) actually depend on.

Reply:
I'll be happy to work on this.